### PR TITLE
accept base64 data and convert it to binary

### PR DESCRIPTION
**Pull request by anoskov**
Extend Arc.File.new/1 function for accepts a data in base64 format.

resolves "#107"

### CODE REVIEW

File: lib/arc/file.ex [hunk 1]
```diff
@@ -20,6 +20,12 @@ defmodule Arc.File do
     end
   end
 
+  # Accepts a data in base64
+  def new(%{filename: filename, base64: base64}) do
+    {:ok, binary} = Base.decode64(base64)
```
Comment (gullitmiranda) [inline lib/arc/file.ex]: I believe that it is interesting to add the option `ignore: :whitespace`
https://github.com/elixir-lang/elixir/issues/4186

**Comment by andrewtimberlake** [comment]
Why not just do `Definition.store(%{filename: "image.png", binary: Base.decode64(data)})`
That way you can handle your own decoding options.

**Comment by anoskov** [comment]
@andrewtimberlake because i'm using arc_ecto and my way allows just pass base64 image to cast_attachments in changeset without workaround.

**Comment by andrewtimberlake** [comment]
@anoskov Ok, I don’t use arc_ecto.

**Comment by stavro** [comment]
I'm not convinced that this is necessary enough to include in `Arc`.

You can do the same thing with arc_ecto:

```elixir
user = Blvd.Repo.one!(Blvd.User.Schema)

params = %{
  avatar: %{
    filename: "image.png",
    binary: Base.decode64!("iVBORw0KGgoAAAANSUhEUgAAABgAAAAYCAMAAADXqc3KAAAB+FBMVEUAAAA/mUPidDHiLi5Cn0XkNTPmeUrkdUg/m0Q0pEfcpSbwaVdKskg+lUP4zA/iLi3msSHkOjVAmETdJSjtYFE/lkPnRj3sWUs8kkLeqCVIq0fxvhXqUkbVmSjwa1n1yBLepyX1xxP0xRXqUkboST9KukpHpUbuvRrzrhF/ljbwaljuZFM4jELaoSdLtElJrUj1xxP6zwzfqSU4i0HYnydMtUlIqUfywxb60AxZqEXaoifgMCXptR9MtklHpEY2iUHWnSjvvRr70QujkC+pUC/90glMuEnlOjVMt0j70QriLS1LtEnnRj3qUUXfIidOjsxAhcZFo0bjNDH0xxNLr0dIrUdmntVTkMoyfL8jcLBRuErhJyrgKyb4zA/5zg3tYFBBmUTmQTnhMinruBzvvhnxwxZ/st+Ktt5zp9hqota2vtK6y9FemNBblc9HiMiTtMbFtsM6gcPV2r6dwroseLrMrbQrdLGdyKoobKbo3Zh+ynrgVllZulTsXE3rV0pIqUf42UVUo0JyjEHoS0HmsiHRGR/lmRz/1hjqnxjvpRWfwtOhusaz0LRGf7FEfbDVmqHXlJeW0pbXq5bec3fX0nTnzmuJuWvhoFFhm0FtrziBsjaAaDCYWC+uSi6jQS3FsSfLJiTirCOkuCG1KiG+wSC+GBvgyhTszQ64Z77KAAAARXRSTlMAIQRDLyUgCwsE6ebm5ubg2dLR0byXl4FDQzU1NDEuLSUgC+vr6urq6ubb29vb2tra2tG8vLu7u7uXl5eXgYGBgYGBLiUALabIAAABsElEQVQoz12S9VPjQBxHt8VaOA6HE+AOzv1wd7pJk5I2adpCC7RUcHd3d3fXf5PvLkxheD++z+yb7GSRlwD/+Hj/APQCZWxM5M+goF+RMbHK594v+tPoiN1uHxkt+xzt9+R9wnRTZZQpXQ0T5uP1IQxToyOAZiQu5HEpjeA4SWIoksRxNiGC1tRZJ4LNxgHgnU5nJZBDvuDdl8lzQRBsQ+s9PZt7s7Pz8wsL39/DkIfZ4xlB2Gqsq62ta9oxVlVrNZpihFRpGO9fzQw1ms0NDWZz07iGkJmIFH8xxkc3a/WWlubmFkv9AB2SEpDvKxbjidN2faseaNV3zoHXvv7wMODJdkOHAegweAfFPx4G67KluxzottCU9n8CUqXzcIQdXOytAHqXxomvykhEKN9EFutG22p//0rbNvHVxiJywa8yS2KDfV1dfbu31H8jF1RHiTKtWYeHxUvq3bn0pyjCRaiRU6aDO+gb3aEfEeVNsDgm8zzLy9egPa7Qt8TSJdwhjplk06HH43ZNJ3s91KKCHQ5x4sw1fRGYDZ0n1L4FKb9/BP5JLYxToheoFCVxz57PPS8UhhEpLBVeAAAAAElFTkSuQmCC")
  }
}

user
|> Blvd.User.Schema.update_avatar(params)
|> Blvd.Repo.update!()
```

**Pull request closed without merging**